### DISCUSSION TITLE
Lombok not working

### DISCUSSION THREAD

**Discussion by U239087** [mfussenegger/nvim-jdtls#625] [Q&A] [answered]
I recently tried installing nvim-jdtls and everything seems to be working fine now except for lombok. 
All the methods have the "The method xxx is undefined for the type yyy" diagnostic error.

How I set up my jdtls:

<img width="440" alt="image" src="https://github.com/mfussenegger/nvim-jdtls/assets/147405637/ac49f24b-d907-483c-8a1e-41e8aec2daf4">

The lombok path equates to '\~/.local/share/nvim/mason/packages/jdtls/lombok.jar'. I also already tried just downloading the lombok.jar from the website (snapshot version 1.18.31 and older versions like 1.18.0) and setting the value to '\~/Downloads/lombok-edge.jar' and '~/Downloads/lombok-1.18.0.jar' respectively. All of that didn't work.

When executing ps aux | grep 'javaagent' I can see that the command gets executed with the correct path.
I use [@pavva91](https://github.com/pavva91)s configuration from the wiki, just modified to work with a m3 macbook.

I also use mason, which I had to configure jdtls to specifically start on filetype java or else nothing would happen.
And like I said, everything else seems to be working fine.

Any ideas on why lombok doesn't want to work?

**Comment by U239087** [comment, marked as answer]
Update: I realized that I was being stupid. I supplied the wrong jar file to be opened.
Problem is now that even less is working than before. The lsp starts "correctly?" but not only are the lombok errors still here but now not even String can be resolved to a type.
If this is a common problem (I assume im still having a major skill issue) I would appreciate some helpful inputs as to what the problem could be or how I can debug this.

**Reply by nramirezuy** [reply]
I might have the same skill issue, because my lombok is still failing as well

**Reply by U239087** [reply]
Just to explain as to why I marked this as the answer. The fact that I had used the wrong path was indeed the error but even after fixing that it didn't seem to do anything. But after restarting everything, the errors resolved. So for future readers I suggest restarting neovim frequently and maybe even the pc once in a while, can't really say more since this was a year ago. Everything works fine now.

**Reply by fromSmolsoft** [reply]
Yep good old "restart" worked for me as well just minute ago when Lombok suddenly went "absent" after refactoring.
(just leaving it here for future readers.)
